advanced_message_passing.md

```diff
@@ -1,8 +1,120 @@
+
+##### Privacy Considerations
+
+Any information passed via a URL or QR code is unencrypted, and may be observed by another party. This lack of privacy must be minded in two different ways.
+
+First, no private information may be passed in the message. Private information should be passed between parties in encrypted messages only. Any protocol message that contains private information should not be passed via URL or QR code.
+
+Second, any identifiers passed in a message sent via URL or QR code must no longer be considered private. Any DID used or other identifier no longer considered private MUST be rotated over a secure connection if privacy is required.
+
+##### Standard Message Encoding
+
+Using a standard message encoding allows for easier interoperability between multiple projects and software platforms. Using a URL for that standard encoding provides a built in fallback flow for users who are unable to automatically process the message. Those new users will load the URL in a browser as a default behavior, and may be presented with instructions on how to install software capable of processing the message. Already onboarded users will be able to process the message without loading in a browser via mobile app URL capture, or via capability detection after being loaded in a browser.
+
+The standard message format is a URL with a Base64URLEncoded plaintext JWM json object as a query parameter.
+
+The URL format is as follows, with some elements described below:
+
+```text
+https://<domain>/<path>?_dc=<encodedplaintextjwm>
+```
+
+`<domain>` and `<path>` should be kept as short as possible, and the full URL should return human readable instructions when loaded in a browser. This is intended to aid new users. The `_dc` query parameter is required and is reserved to contain the DIDComm message string. Additional path elements or query parameters are allowed, and can be leveraged to provide coupons or other promise of payment for new users.
+
+> To do: We need to rationalize this approach `https://` approach with the use of a special protocol (e.g. `didcomm://`) that will enable handling of the URL on mobile devices to automatically invoke an installed app on both Android and iOS. A user must be able to process the out-of-band message on the device of the agent (e.g. when the mobile device can't scan the QR code because it is on a web page on device).
+
+The `<encodedplaintextjwm>` is a JWM plaintext message that has been base64-url encoded.
+
+```javascript
+encodedplaintextjwm = b64urlencode(<plaintextjwm>)
+```
+
+During encoding, whitespace from the json string should be eliminated to keep the resulting out-of-band message string as short as possible.
+
+##### Example Out-of-Band Message Encoding
+
+Invitation:
```

Comment:
@troyronda @llorllale Does adding this address your concerns about a common invitation format?